### PR TITLE
[MiniZinc] add a builder for MiniZinc

### PR DESCRIPTION
**Pull request by odow**

### CODE REVIEW

File: M/MiniZinc/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+]
+
+script = raw"""
+cd $WORKSPACE/srcdir
+
+mkdir -p libminizinc/build
+cd libminizinc/build
+
+# Patch for MinGW toolchain
+find . -type f -exec sed -i 's/Windows.h/windows.h/g' {} +
```
Comment (odow) [inline M/MiniZinc/build_tarballs.jl:23]:
```suggestion
find .. -type f -exec sed -i 's/Windows.h/windows.h/g' {} +
```

**Comment by odow** [comment]
So many string issues on Windows. This JLL is going to be the base of a large-ish Julia package I need to write, so I wonder if we can merge without Windows support. Then I can get things implemented and tested on Mac and linux, and come back to Windows support before I release MiniZinc.jl as a public package.

File: M/MiniZinc/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+make install
+"""
+
+products = [
+    ExecutableProduct("minizinc", :minizinc),
+]
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
```
Comment (giordano) [inline M/MiniZinc/build_tarballs.jl:51]:
```suggestion
platforms = expand_cxxstring_abis(supported_platforms())
```

**Comment by odow** [comment]
Thanks, as always, @giordano

**Pull request merged**